plugin/storage/es/factory.go

```diff
@@ -312,8 +312,12 @@ func (f *Factory) onClientPasswordChange(cfg *config.Configuration, client *atom
 	if err != nil {
 		f.logger.Error("failed to recreate Elasticsearch client with new password", zap.Error(err))
 	} else {
-		client.Store(&primaryClient)
+		oldClient := client.Swap(&newClient)
```

Comment:
yep, exactly what I had in mind. It would be good to also add this to factory.Close(), i.e. swap both primary and archive clients for nil and close them

Reply:
should I change this to primary client?
oldClient := client.Swap(primaryClient)
then close the old client

Reply:
no, I have no problem with this code. One improvement you could make is rename primaryClient->newClient.

I was referring to adding Close of clients to Factory.Close() method. Right now we leave them hanging.